packages/next/src/client/components/app-router.tsx

```diff
@@ -223,11 +223,7 @@ function Router({
   )
 
   const navigate: RouterNavigate = useCallback(
-    (
-      href: string,
-      navigateType: 'push' | 'replace',
-      forceOptimisticNavigation: boolean
-    ) => {
+    (href, navigateType, forceOptimisticNavigation, shouldScroll) => {
```

Comment:
Since the `RouterNavigate` type is added above these parameters don't need to be explicitly typed.